python_modules/libraries/dagster-dg/dagster_dg/cli/utils.py

```diff
@@ -137,6 +137,24 @@ def _workspace_entry_for_project(dg_context: DgContext) -> dict[str, dict[str, s
     return {"python_module": entry}
 
 
+def _semver(version: str) -> Optional[tuple[int, int, int]]:
+    try:
+        return cast(tuple[int, int, int], tuple(map(int, version.split("."))))
+    except ValueError:
+        return None
+
+
+def _semver_less_than(version: str, other: str) -> bool:
+    version_semver = _semver(version)
+    other_semver = _semver(other)
+    if version_semver is None or other_semver is None:
+        return False
+    return version_semver < other_semver
```

Comment:
use packaging.version.parse for this instead - there is lots of prior art, see e.g. https://github.com/dagster-io/dagster/blob/1d615c48cf5fd3c46ba2237c803d6de946081da2/python_modules/dagster/dagster/_core/remote_representation/feature_flags.py#L44-L48

make sure it behaves currently on local dev